source/slang/parser.cpp

```diff
@@ -2089,6 +2109,41 @@ namespace Slang
         }
     }
 
+    RefPtr<RefObject> ParseAssocType(Parser * parser, void *)
+    {
+        RefPtr<AssocTypeDecl> assocTypeDecl = new AssocTypeDecl();
+
+        auto nameToken = parser->ReadToken(TokenType::Identifier);
+        assocTypeDecl->nameAndLoc = NameLoc(nameToken);
+        assocTypeDecl->loc = nameToken.loc;
+        if (AdvanceIf(parser, TokenType::Colon))
+        {
+            while (!parser->tokenReader.IsAtEnd())
+            {
+                auto paramConstraint = new GenericTypeConstraintDecl();
```

Comment:
I don’t recall the solution the parser uses, but we need to be careful about any loop like this that could go into an infinite loop if we have a syntax error.